quill-orientdb/src/main/scala/io/getquill/context/orientdb/OrientDBIdiom.scala

```diff
@@ -119,13 +119,13 @@ trait OrientDBIdiom extends Idiom {
   implicit def orientDBQueryTokenizer(implicit strategy: NamingStrategy): Tokenizer[SqlQuery] = Tokenizer[SqlQuery] {
     case FlattenSqlQuery(from, where, groupBy, orderBy, limit, offset, select, distinct) =>
 
-      val distinctTokenizer = (if (distinct) "DISTINCT" else "").token
+      val distinctTokenizer = (if (distinct == DistinctKind.Distinct) "DISTINCT" else "").token
 
       val selectClause =
         select match {
-          case Nil => if (!distinct) stmt"SELECT *" else fail("OrientDB DISTINCT with multiple columns is not supported")
```

Comment:
This should also fail if it's DistinctOn regardless of any criteria, what's the best way to do that?